pkg/gsheets/sheet.go

```diff
@@ -0,0 +1,185 @@
+		"release-openshift-ocp-installer-e2e-openstack-4.3":           1408408210,
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.3":    1992493886,
+		"release-openshift-ocp-installer-e2e-openstack-4.2":           493400895,
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.2":    254644539,
+		"release-openshift-origin-installer-e2e-openstack-4.2":        61282267,
+		"release-openshift-origin-installer-e2e-openstack-serial-4.2": 1799792403,
+	}
+
+	sheetId, ok := innerMap[s.JobName]
+	if !ok {
+		log.Fatalf("Unknown job %v", s.JobName)
+	}
+
+	return sheetId
+}
+
+func (s *Sheet) getName() string {
+	innerMap := map[string]string{
+		"release-openshift-ocp-installer-e2e-openstack-4.4":           "Parallel OCP 4.4",
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.4":    "Serial OCP 4.4",
+		"release-openshift-ocp-installer-e2e-openstack-4.3":           "Parallel OCP 4.3",
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.3":    "Serial OCP 4.3",
+		"release-openshift-ocp-installer-e2e-openstack-4.2":           "Parallel OCP 4.2",
+		"release-openshift-ocp-installer-e2e-openstack-serial-4.2":    "Serial OCP 4.2",
+		"release-openshift-origin-installer-e2e-openstack-4.2":        "Parallel Origin 4.2",
+		"release-openshift-origin-installer-e2e-openstack-serial-4.2": "Serial Origin 4.2",
+	}
+
+	sheetName, ok := innerMap[s.JobName]
+	if !ok {
+		log.Fatalf("Unknown job %v", s.JobName)
+	}
+
+	return sheetName
+}
+
+func (s *Sheet) getAllIDs() {
+	readRange := s.getName() + "!A7:A300"
+	resp, err := s.Client.service.Spreadsheets.Values.Get(s.Client.spreadsheetId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v", err)
+	}
+
+	if len(resp.Values) == 0 {
+		fmt.Println("No data found.")
+		return
+	}
+
+	for _, row := range resp.Values {
+		id, err := strconv.Atoi(fmt.Sprint(row[0]))
+		if err != nil {
+			id = 0
+		}
+		s.idxList = append(s.idxList, id)
+	}
+}
+
+// Get the index on which the job with the given id should be inserted
+func (s *Sheet) getIndex(id int) (int, bool) {
+	// Populate list of IDs from sheet
+	if len(s.idxList) == 0 {
+		s.getAllIDs()
+	}
+
+	i := 0
+	for ; i < len(s.idxList); i++ {
+		// s.idxList is sorted in descending order
+		if s.idxList[i] <= id {
+			break
+		}
+	}
+
+	exists := len(s.idxList) > 0 && s.idxList[i] == id
+
+	return i, exists
+}
+
+// Get ID of the most recent job in the spreadsheet
+func (s *Sheet) GetLatestId() int {
+	readRange := s.getName() + "!A7"
+	resp, err := s.Client.service.Spreadsheets.Values.Get(s.Client.spreadsheetId, readRange).Do()
+	if err != nil {
+		log.Fatalf("Unable to retrieve data from sheet: %v", err)
+	}
+
+	if len(resp.Values) == 0 || len(resp.Values[0]) == 0 {
+		return 0
+	}
+
+	data := fmt.Sprint(resp.Values[0][0])
+	id, err := strconv.Atoi(data)
+	if err != nil {
+		return 0
+	}
+
+	return id
+}
+
+func (s *Sheet) AddRow(j job.Job) {
```

Comment:
Watch out, that's a circular reference:
* `gsheets` knows about `job`, because it needs both job-id and job-html;
* `job` knows about the spreadsheet: the XML logic of `ToHtml` intrinsically does.

You could move `ToHtml` to this package. That way, `job` stays focused on the CI domain; while here we deal with the spreadsheet.

As a side effect, this solution fixes your `Username` problem: it's fine for `gsheets` to know about it.

Reply:
For bonus points, apply dependency inversion the Go way!

You can give `job.Job` a bunch of exported methods, and then consume them through an interface in `gsheet`